server/controllers/admin/nationalOccupancyController.ts

```diff
@@ -0,0 +1,88 @@
+import type { Request, Response, TypedRequestHandler } from 'express'
+import { Cas1SpaceBookingCharacteristic, Cas1SpaceCharacteristic } from '@approved-premises/api'
+import { NationalSpaceSearchFormData } from '@approved-premises/ui'
+import { CruManagementAreaService } from '../../services'
+import { DateFormats, isoDateIsValid } from '../../utils/dateUtils'
+import { getApTypeOptions, getManagementAreaSelectGroups } from '../../utils/admin/nationalOccupancyUtils'
+import { convertKeyValuePairToCheckBoxItems, validPostcodeArea } from '../../utils/formUtils'
+import { roomCharacteristicMap } from '../../utils/characteristicsUtils'
+import { spaceSearchCriteriaApLevelLabels } from '../../utils/match/spaceSearchLabels'
+import { makeArrayOfType } from '../../utils/utils'
+import { generateErrorMessages, generateErrorSummary } from '../../utils/validation'
+import paths from '../../paths/admin'
+import MultiPageFormManager from '../../utils/multiPageFormManager'
+
+export default class NationalOccupancyController {
+  formData: MultiPageFormManager<'nationalSpaceSearch'>
+
+  constructor(private readonly cruManagementAreaService: CruManagementAreaService) {
+    this.formData = new MultiPageFormManager('nationalSpaceSearch')
+  }
+
+  index(): TypedRequestHandler<Request, Response> {
+    return async (req: Request, res: Response) => {
+      const {
+        locals: {
+          user: { cruManagementArea },
+        },
+      } = res
+      const {
+        user: { token },
+        query,
+      } = req
+
+      const sessionData = this.formData.get('', req.session) || {}
+      const combinedQuery = { ...sessionData, ...query }
+
+      const {
+        postcode,
+        arrivalDate: rawArrivalDate,
+        apArea,
+        apType,
+        roomCriteria,
+        apCriteria,
+      } = combinedQuery as NationalSpaceSearchFormData
+
+      const errors: Record<string, string> = {}
+
+      const slashToday = DateFormats.isoDateToUIDate(DateFormats.dateObjToIsoDate(new Date()), { format: 'datePicker' })
+
+      const cruManagementAreas = await this.cruManagementAreaService.getCruManagementAreas(token)
+      const arrivalDate = rawArrivalDate || slashToday
+
+      if (!isoDateIsValid(DateFormats.datepickerInputToIsoString(arrivalDate as string))) {
+        errors.arrivalDate = 'Enter a valid arrival date'
+      }
+      if (postcode && !validPostcodeArea(postcode)) {
+        errors.postcode = 'Enter a valid postcode area'
+      }
+
+      if (!errors.length) {
+        await this.formData.update('', req.session, combinedQuery)
```

Comment:
This effectively creates an empty key in the `nationalSpaceSearch` session data, i.e. `{ '': combinedQuery }`, which is valid, but a bit odd -- maybe give it the key `default` or something similarly generic?

Reply:
OK, yes It is a bit odd, but I thought better to re-use the existing structure. A default key makes sense.